org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProviderTest.java

```diff
@@ -100,6 +100,12 @@ public void testFixAllEmptyEnumLiteral() throws Exception {
 		assertAndApplyAllResolutions(editorForGrammar("Model: a=ID;", "enum ABCD: A|B|C=''|D;", "enum B: B='';"), EMPTY_ENUM_LITERAL, 1, 2,
 				"Fix empty enum literal");
 	}
+	
+	@Test
+	public void testFixKeywordNoSpaces() throws Exception {
+		assertAndApplySingleResolution(editorForGrammar("Model: ' a b c d ' a=ID;"), SPACES_IN_KEYWORD, 0,
```

Comment:
What happens, if the keyword with spaces is used in an enum rule?

Reply:
The current validation check in [XtextValidator](https://github.com/eclipse/xtext-core/blob/79c99fd5d24d3875441b8ba419bfd06857af07b3/org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java#L1089) only considers parser rule keywords. This quickfix is not available for keywords with spaces in enum rules. Should we extend the validation check  to enum rules and define a dedicated quickfix replacing all spaces into a single identifier?

Reply:
No, I don't think that this would be possible with reasonable effort - the ecore model, that is generated from enum rules, relies on the fact that the rhs of an enum literal declaration is a single keyword.